lib/client.js

```diff
@@ -467,7 +467,14 @@ function Request(method, url) {
       self.emit('response', res);
     }
 
-    self.callback(err, res);
+    if (res && res.status >= 200 && res.status < 300) {
+      return self.callback(err, res);
+    }
+
+    var new_err = new Error('Unsucessful HTTP response');
```

Comment:
I think that bringing the object `exports.STATUS_CODES` into the `superagent` library will aggregate value, since that the map object of status codes is for a stable protocol.

How can I treat the error? How can I know if that error is a transport error or an HTTP Error?

I think we could create a `new HTTPError('<code> "message"')` and maybe a `NetworkError` too. So, by example: we can retry the request if the error is a `NetworkError`.